test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -157,12 +155,54 @@ TEST(NoProxyTest, WhitespacedEntries) {
 
 // Test entries with invalid CIDR blocks in 'no_proxy' list.
 TEST(NoProxyTest, InvalidCIDREntries) {
-  ScopedSetEnv no_proxy("foo.com, 192.168.0.255/33");
+  ScopedEnvVar no_proxy(NO_PROXY_VAR_NAME, "foo.com, 192.168.0.255/33");
   auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "http://proxy.google.com");
   EXPECT_EQ(HttpProxyMapper().MapName("dns:///192.168.1.0:443", &args),
             "proxy.google.com");
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.1.0:443");
 }
+
+TEST(ProxyForAddressTest, SetByEnvVar) {
+  ScopedEnvVar address_proxy(HttpProxyMapper::ADDRESS_PROXY_ENV_VAR,
+                             "192.168.0.100:2020");
+  auto args = ChannelArgs();
+  auto address = StringToSockaddr("192.168.0.1:3333");
+  ASSERT_EQ(address.status().code(), absl::StatusCode::kOk);
+  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
+  ASSERT_TRUE(mapped.has_value());
+  EXPECT_EQ(grpc_sockaddr_to_string(&mapped.value(), true),
+            absl::StatusOr<std::string>("192.168.0.100:2020"));
+  EXPECT_EQ(
+      std::string(
+          args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER).value_or("<empty>")),
+      "192.168.0.1:3333");
+}
+
+TEST(ProxyForAddressTest, ChannelArgPreferred) {
+  ScopedEnvVar address_proxy(HttpProxyMapper::ADDRESS_PROXY_ENV_VAR,
+                             "192.168.0.100:2020");
+  auto args = ChannelArgs().Set(GRPC_ARG_ADDRESS_PROXY, "192.168.0.101:2020");
+  auto address = StringToSockaddr("192.168.0.1:3333");
+  ASSERT_EQ(address.status().code(), absl::StatusCode::kOk);
+  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
+  ASSERT_TRUE(mapped.has_value());
+  auto address_string = grpc_sockaddr_to_string(&mapped.value(), true);
+  EXPECT_EQ(address_string.status().code(), absl::StatusCode::kOk);
```

Comment:
This should be `ASSERT` instead of `EXPECT`, because if it's not true, it will crash to access `*address_string` on the next line.

See also my comment above about checking `ok()` instead of the code.

Reply:
Done